wire/addr_test.go

```diff
@@ -68,4 +68,35 @@ var _ = Describe("Address", func() {
 			Expect(quick.Check(f, nil)).To(Succeed())
 		})
 	})
+
+	Context("when verifying addresses", func() {
+		It("should succeed for addresses with valid signatures", func() {
+			f := func() bool {
+				pk := id.NewPrivKey()
+				expectedSignatory := pk.Signatory()
+				wireAddress := wire.NewUnsignedAddress(wire.TCP, "0.0.0.0", uint64(time.Now().UnixNano()))
+				wireAddress.Sign(pk)
+				Expect(wireAddress.Verify(expectedSignatory)).To(Succeed())
+				return true
+			}
+			Expect(quick.Check(f, nil)).To(Succeed())
+		})
+
+		It("should fail for addresses with invalid signatures", func() {
+			f := func() bool {
+				pk := id.NewPrivKey()
+				expectedSignatory := pk.Signatory()
+				wireAddress := wire.NewUnsignedAddress(wire.TCP, "0.0.0.0", uint64(time.Now().UnixNano()))
+				rand.Read(wireAddress.Signature[:])
+				// Half of the time, make sure that the recovery ID for the
+				// signature is valid.
+				if rand.Int()%2 == 1 {
+					wireAddress.Signature[64] %= 4
+				}
+				Expect(wireAddress.Verify(expectedSignatory)).ToNot(Succeed())
```

Comment:
Could be worth checking the error message is as expected for the two cases (e.g. by checking it contains a certain substring).